server/app/services/geo/esri/EsriClient.java

```diff
@@ -262,36 +269,91 @@ CompletionStage<Optional<JsonNode>> fetchServiceAreaFeatures(
       EsriServiceAreaValidationOption esriServiceAreaValidationOption, AddressLocation location) {
     return fetchServiceAreaFeatures(location, esriServiceAreaValidationOption.getUrl())
         .thenApply(
             (maybeJson) -> {
+              EsriServiceAreaInclusion.Builder esriServiceAreaInclusionBuilder =
+                  EsriServiceAreaInclusion.builder();
+              ImmutableList.Builder<EsriServiceAreaInclusion> inclusionListBuilder =
+                  ImmutableList.builder();
               if (maybeJson.isEmpty()) {
                 logger.error(
                     "EsriClient.fetchServiceAreaFeatures JSON response is empty. Called by"
                         + " EsriClient.isAddressLocationInServiceArea."
                         + " EsriServiceAreaValidationOption = {}. AddressLocation = {}",
                     esriServiceAreaValidationOption,
                     location);
-                return Optional.empty();
+                inclusionListBuilder.add(
+                    esriServiceAreaInclusionBuilder
+                        .setArea(esriServiceAreaValidationOption.getId())
+                        .setState(EsriServiceAreaState.FAILED)
+                        .setTimeStamp(Instant.now())
+                        .build());
+                return inclusionListBuilder.build();
+              }
+
+              Optional<ImmutableMap<String, EsriServiceAreaValidationOption>> optionMap =
+                  esriServiceAreaValidationConfig.getImmutableMap();
+
+              if (optionMap.isEmpty()) {
+                logger.error(
+                    "Error calling EsriClient.getServiceAreaInclusionGroups. Error:"
+                        + " EsriServiceAreaValidationConfig.getImmutableMap() returned empty.");
+                esriServiceAreaInclusionBuilder
+                    .setArea(esriServiceAreaValidationOption.getId())
+                    .setState(EsriServiceAreaState.FAILED)
+                    .setTimeStamp(Instant.now());
+                inclusionListBuilder.add(esriServiceAreaInclusionBuilder.build());
+                return inclusionListBuilder.build();
               }
 
+              ImmutableList.Builder<EsriServiceAreaInclusion> listBuilder = ImmutableList.builder();
+
               JsonNode json = maybeJson.get();
               ReadContext ctx = JsonPath.parse(json.toString());
               List<String> features =
                   ctx.read(
                       "features[*].attributes." + esriServiceAreaValidationOption.getAttribute());
-              Optional<String> feature =
-                  features.stream()
-                      .filter(val -> esriServiceAreaValidationOption.getId().equals(val))
-                      .findFirst();
-              return Optional.of(feature.isPresent());
+
+              for (EsriServiceAreaValidationOption option : optionMap.get().values()) {
+                if (features.contains(option.getId())) {
```

Comment:
**suggestion (blocking):** it's a bit confusing that each option has its own `attribute` and `url` when the code here and in the call to ESRI assumes that the attribute and URl will be the same for all of them.

Please restructure the code or add comments explaining the assumptions we make around this.

Reply:
Thanks for catching this. This is updated so that it adds the service areas that have the same URL specified in the config to the inclusion group.

This does mean that CF will have to call this method again if a service area validation predicate is selected that hasn't been checked against (e.i. the service area ID isn't in the inclusion group).

The other option would be to fetch service area results for each unique URL specified. However, in most cases it seems the URL will be the same and we'll have to accommodate adding new service area options to the config and allow for subsequent calls to fetch previously unchecked service areas anyway.